skimage/filters/_frangi.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+
+__all__ = ['frangi', 'hessian']
+
+
+def filtered_image_calculation(image, scale, scale_step, beta1, beta2,
+                               frangi_=True, black_ridges=True):
+    """This is an intermediate function for Frangi and Hessian filters.
+
+    Shares the common code for frangi and hessian functions.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    beta1 : float, optional
+        Frangi correction constant.
+    beta2 : float, optional
+        Frangi correction constant.
+    black_ridges : boolean, optional
+        Detect black ridges (default) set to true, for
+        white ridges set to false.
+
+    Returns
+    -------
+    filtered_list : list
+        List of pre-filtered images.
+
+    """
+
+    # Import has to be here due to circular import error
+    from ..feature import hessian_matrix, hessian_matrix_eigvals
+
+    sigmas = np.arange(scale[0], scale[1], scale_step)
+
+    if np.any(np.asarray(sigmas) < 0.0):
+        raise ValueError("Sigma values less than zero are not valid")
+
+    beta1 = 2 * beta1 ** 2
+    beta2 = 2 * beta2 ** 2
+
+    # Make matrices to store all filtered images
+    filtered_list = []
+
+    # Frangi filter for all sigmas
+    for sigma in sigmas:
+
+        # Make 2D hessian
+        (Dxx, Dxy, Dyy) = hessian_matrix(image, sigma)
+
+        # Correct for scale
+        Dxx = (sigma ** 2) * Dxx
+        Dxy = (sigma ** 2) * Dxy
+        Dyy = (sigma ** 2) * Dyy
+
+        # Calculate (abs sorted) eigenvalues and vectors
+        (Lambda1, Lambda2) = hessian_matrix_eigvals(Dxx, Dxy, Dyy)
+
+        # Compute some similarity measures
+        Lambda1[np.where(Lambda1 == 0)] = 1e-10
+        Rb = (Lambda2 / Lambda1) ** 2
+        S2 = Lambda1 ** 2 + Lambda2 ** 2
+
+        # Compute the output image
+        filtered = np.exp(-Rb / beta1) * (np.ones(np.shape(image)) -
+                                          np.exp(-S2 / beta2))
+        if frangi_:
```

Comment:
@Hiyorimi I would still consider that preferable, actually. This function could easily have that interface since it is private. Then the pattern is:

- run common filter
- get back filtered images and lambda values
- do something different with output depending on the desired filter.

I think that's cleaner. =)